File: apps/kidcode-studio/src/types/project.ts
import type { Block } from './blocks';

export type ProjectMode = 'drawing' | 'story' | 'puzzle' | 'music';

export interface Project {
    id: string;
    title: string;
    mode: ProjectMode;
    blocks: Block[];
    assets?: ProjectAssets;
    createdAt: Date;
    updatedAt: Date;
}

export interface ProjectAssets {
    background?: string;
    character?: string;
    sounds?: string[];
}

export interface CreateProjectDto {
    title: string;
    mode: ProjectMode;
    blocksJson?: string;
    assets?: string;
}

export interface UpdateProjectDto {
    title?: string;
    blocksJson?: string;
    assets?: string;
}

File: apis/KidCode.Api/Models/Project.cs
namespace KidCode.Api.Models;

public class Project
{
    public Guid Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public ProjectMode Mode { get; set; }
    public string BlocksJson { get; set; } = "[]";
    public string? ModeDataJson { get; set; }
    public string? Assets { get; set; }
    public bool IsDraft { get; set; } = false;
    public string? UserId { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? LastAutoSaveAt { get; set; }

    public ICollection<MediaAsset> MediaAssets { get; set; } = new List<MediaAsset>();
}

public enum ProjectMode
{
    Drawing,
    Story,
    Puzzle,
    Music,
}
